crates/sparrow-execution/src/tests.rs

```diff
@@ -0,0 +1,97 @@
+        Field::new("c", DataType::Int64, true),
+    ]));
+    let source_type = DataType::Struct(source_schema.fields().clone());
+    let source_uuid = session
+        .add_source(source_schema.clone(), sparrow_session::SourceConfig)
+        .unwrap();
+
+    let source = Arc::new(sparrow_logical::Expr::new_uuid(
+        "read",
+        source_uuid,
+        source_type,
+        sparrow_logical::Grouping::new(0),
+    ));
+    let a_str = Arc::new(sparrow_logical::Expr::new_literal_str("a"));
+    let a =
+        Arc::new(sparrow_logical::Expr::try_new("fieldref", vec![source.clone(), a_str]).unwrap());
+    let b_str = Arc::new(sparrow_logical::Expr::new_literal_str("b"));
+    let b =
+        Arc::new(sparrow_logical::Expr::try_new("fieldref", vec![source.clone(), b_str]).unwrap());
+    let a_plus_b = Arc::new(sparrow_logical::Expr::try_new("add", vec![a, b]).unwrap());
+
+    let c_str = Arc::new(sparrow_logical::Expr::new_literal_str("c"));
+    let c =
+        Arc::new(sparrow_logical::Expr::try_new("fieldref", vec![source.clone(), c_str]).unwrap());
+    let a_plus_b_plus_c =
+        Arc::new(sparrow_logical::Expr::try_new("add", vec![a_plus_b.clone(), c]).unwrap());
+
+    let ab_str = Arc::new(sparrow_logical::Expr::new_literal_str("ab"));
+    let abc_str = Arc::new(sparrow_logical::Expr::new_literal_str("abc"));
+    let ab_and_abc = Arc::new(
+        sparrow_logical::Expr::try_new("record", vec![ab_str, a_plus_b, abc_str, a_plus_b_plus_c])
+            .unwrap(),
+    );
+
+    let plan = sparrow_backend::compile(&ab_and_abc, None).unwrap();
+    print!("Plan: {}", plan.yaml());
+
+    let (input_tx, input_rx) = tokio::sync::mpsc::channel(10);
+    let (output_tx, mut output_rx) = tokio::sync::mpsc::channel(10);
+
+    let mut inputs = HashMap::with_capacity(1);
+    inputs.insert(source_uuid, input_rx);
+    let executor =
+        crate::PlanExecutor::try_new("test_query_id".to_owned(), plan, inputs, output_tx).unwrap();
+
+    // Before running, make sure there is an input batch available.
+    // Then, drop the `input_tx` so the channel closes.
+    let input_data = Arc::new(StructArray::new(
+        source_schema.fields().clone(),
+        vec![
+            Arc::new(Int64Array::from(vec![0, 1, 2, 3])),
+            Arc::new(Int64Array::from(vec![4, 7, 10, 11])),
+            Arc::new(Int64Array::from(vec![Some(21), None, Some(387), Some(87)])),
+        ],
+        None,
+    ));
+    let time = Arc::new(TimestampNanosecondArray::from(vec![0, 1, 2, 3]));
+    let subsort = Arc::new(UInt64Array::from(vec![0, 1, 2, 3]));
+    let key_hash = Arc::new(UInt64Array::from(vec![0, 1, 2, 3]));
+    let input_batch = Batch::new_with_data(
+        input_data,
+        time,
+        subsort,
+        key_hash,
+        RowTime::from_timestamp_ns(3),
+    );
+    input_tx.blocking_send(input_batch).unwrap();
+    std::mem::drop(input_tx);
+
+    session.rt.block_on(executor.execute()).unwrap();
```

Comment:
I don't like this.

Originally the test was `tokio::test` (which meant it got a runtime), but then the runtime inside the created session was getting dropped.

I think there are two options:

1. Codify something about what is done here -- instead of `session.rt.block_on(executor.execute())`, I can just add the code to session to take a `sparrow_logical::Expr` and execute it. Then I don't need to expose the runtime here. That is probably the best option.
2. Add a method to create a session using a specific runtime or runtime handle. Then pass in the current runtime to the session. Don't love that due to the unclear dropping of the runtime.